x-pack/plugins/observability/server/services/slo/summary_cleanup_task.ts

```diff
@@ -0,0 +1,167 @@
+
+                for await (const response of finder.find()) {
+                  const soItems = response.saved_objects.map((so) => so.attributes.id);
+
+                  const { sloDefIds, searchAfter } = await this.fetchSloDefinitions(searchAfterKey);
+                  searchAfterKey = searchAfter;
+
+                  const { missingSloDefIds, missingSOIds } = findMissingItems(
+                    soItems.concat(soIds),
+                    sloDefIds.concat(sloDefIdsToDelete)
+                  );
+
+                  soIds.push(...missingSOIds);
+                  sloDefIdsToDelete = missingSloDefIds;
+                }
+                this.logger.info(
+                  JSON.stringify({
+                    idsToDelete: Array.from(new Set(sloDefIdsToDelete)),
+                    soIds: Array.from(new Set(soIds)),
+                  })
+                );
+
+                if (sloDefIdsToDelete.length > 0) {
+                  await this.esClient.deleteByQuery({
+                    index: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+                    query: {
+                      terms: {
+                        'slo.id': sloDefIdsToDelete,
+                      },
+                    },
+                  });
+                }
+              }
+
+              return { state: { lastRunAt: runAt } };
+            },
+
+            cancel: async () => {
+              this.abortController.abort('[SLO] Definitions clean up Task timed out');
+            },
+          };
+        },
+      },
+    });
+  }
+
+  fetchSloDefinitions = async (searchAfter?: SortResults) => {
+    if (this.esClient) {
+      const result = await this.esClient.search({
+        size: 1000,
+        index: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+        fields: ['slo.id'],
+        collapse: {
+          field: 'slo.id',
+        },
+        _source: false,
+        sort: [
+          {
+            'slo.id': {
+              order: 'desc',
+            },
+          },
+        ],
+        search_after: searchAfter,
+      });
+
+      const newSearchAfter = result.hits.hits[result.hits.hits.length - 1].sort;
+      return {
+        searchAfter: newSearchAfter,
+        sloDefIds: result.hits.hits.map((hit) => hit.fields?.['slo.id'][0] as string),
+      };
+    } else {
+      return {
+        searchAfter: undefined,
+        sloDefIds: [] as string[],
+      };
+    }
+  };
+
+  private get taskId() {
+    return `${TASK_TYPE}:1.0.0`;
+  }
+
+  public async start(
+    taskManager: TaskManagerStartContract,
+    soClient: SavedObjectsClient,
+    esClient: ElasticsearchClient
+  ) {
+    this.taskManager = taskManager;
+    this.soClient = soClient;
+    this.esClient = esClient;
+
+    if (!taskManager) {
+      this.logger.info('[SLO] Missing required service during startup, skipping task.');
+    }
+
+    this.taskManager.ensureScheduled({
+      id: this.taskId,
+      taskType: TASK_TYPE,
+      schedule: {
+        interval: '1m',
```

Comment:
I would run this less frequently, maybe hourly? It is quite an intensive task (going over all the slo), and I don't expect to have many orphans SLOs in summary. You probably need to change the timeout accordingly as well.